sql/catalyst/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceV2Relation.scala

```diff
@@ -80,7 +80,7 @@ case class DataSourceV2Relation(
         s"BUG: computeStats called before pushdown on DSv2 relation: $name")
     } else {
       // when not testing, return stats because bad stats are better than failing a query
-      table.asReadable.newScanBuilder(options) match {
+      table.asReadable.newScanBuilder(options).build() match {
```

Comment:
We always convert `DataSourceV2Relation` to `DataSourceV2ScanRelation` during Optimizer phase, so it should be never used in normal usage. But in case someone use it at Analyzer phase ..